### DISCUSSION TITLE
USD path in Shared Drive

### DISCUSSION THREAD

**Discussion by DGaggio** [Autodesk/maya-usd#2973] [Q&A] [answered]
Hello everyone!

I'm developing a pipeline for my Master at uni around USD. Me and my group want to work in remote on a single Maya scene with USD stage on a Google Drive shared drive.
The problem I'm having is everytime I'm setting up the references between Layers and subLayers and I'm saving the .ma file, if someone try to open it get just an empty scene with an anonymousLayer1 while on my pc it's working properly.
Is there anyway to reference the usd files inside the project folder as we do for textures when we set a project? In this way everyone that set the project would have the Maya file with the USD Layers linked correctly and ready to work.
I've tried to create a new project and in the Project Window there is the possibility to add a new file rule that I called USD same as the folder. Have all my USD files there, when I'm trying to open the project on another pc, it give me the same result, empty scene and an anonymousLayer1. I've also tried to have the usd files on scene folder or data folder but nothing it's working.

Any help would be really appreciate because I don't know what else to try.

Cheers,
Davide

![empty](https://user-images.githubusercontent.com/115561317/228055008-0ed6ac42-4cab-4325-a0ee-eee77ac1a606.png)
![scene](https://user-images.githubusercontent.com/115561317/228055023-4c0dbf3d-cd58-4bd9-8f31-3efe59675135.png)
![master code](https://user-images.githubusercontent.com/115561317/228055038-f7bfb345-de47-41ad-be68-805d3f3c67ac.png)

**Comment by gelshoffNV** [comment]
Does everyone have the same g drive? Seems like as long as you use full paths, things should work...so if your Master_LayerShape points to g:/Shared drives/FMP/Scene/Layers/Master_Layer.usd, it should work for you and everyone. But it does look like using relative paths to your project won't work, neither will settings environment variables like $WORKSPACE/Scene/Layers/Folders/Master_Layer.usd...the env var seems to work on read, but when the scene is written, maya-usd is expanding the path and losing the var, which should be a bug.

**Reply by DGaggio** [reply]
Yes, everyone has the same G driver and it doesn't work but still there is a problem because one of us use Mac that doesn't use drive letters. When I was running some test I was moving the scene folder between my desktop and laptop that has the same username and the path was recognized without any problem.
I'll try to set the variable but it seam a solution specific for this project, is there anything I can set in Maya in general? It would be perfect to have a relative path inside the project folder.

**Comment by womanyoyoyo-adsk** [comment]
Hi @DGaggio,

In [MayaUSD 0.22.0](https://github.com/Autodesk/maya-usd/releases/tag/v0.22.0) we've implemented relative file support for loading and saving the USD root file. That means if you save your scene file in the 'scenes' directory of your Maya project, then you are able to load/save your USD root file (Master_Layer.usd) in this directory as well (or anywhere else in your project) and have the path be relative from the scene file to the USD root file. When loading or creating a USD root file you should get a checkbox like this: 
![image](https://user-images.githubusercontent.com/49446314/228260558-77149bb7-bfb0-46fc-8d32-a93b7df4da35.png)
As of today, the sublayers to be relative to their parent layer is not supported but you can do this manually. So I would put all the sublayers in the same directory as Master_Layer.usd and manually edit the paths of your sublayer in Master_Layer.usd like this:
```
subLayers = [
	@Layout_Folder.usd@,
	@FX_Folder.usd@,
	@Animation_Folder.usd@,
	@Cameras.usd@
]
```
The absolute paths should work though. I am not sure why your colleague opens the .ma file and gets an anonymous file as the root USD layer. Feel free to share this .ma file and I can take a look (the USD files aren't necessary I just need to see the file paths). 

Natalia

**Reply by DGaggio** [reply]
Hi Natalia, 

I updated the plugin and I can see the tick box when I create the stage from usd file but I can't see Unresolved Path and Resolved Path.
I've tried to create the master and edit the path you show in the example but it doesn't work, keep give me an anonymous file.
In the attach zip there is the test with the new plugin version and a folder called Scene that is how I setted up previously and it was working for me.
I'm pretty sure I've made some mistake in the code but I'm new in USD.

Thank you very much for your help.

Davide

[USD_Issue.zip](https://github.com/Autodesk/maya-usd/files/11091112/USD_Issue.zip)

**Comment by womanyoyoyo-adsk** [comment]
Hi @DGaggio,

Good that you see the checkbox, I took a look in your .ma file and the checkbox was used correctly. The problem was syntax errors in all your USD files - brackets missing and instead of 'over' you had 'open'. I've fixed them for you, see attached. Also, if you open the 'Script Editor' in Maya when you have a problem, you can see we identify the problem (syntax error), the USD file, and the line number in the file.

Let me know if you have any other issues. Good luck on your assignment!

Natalia  

[Layout_Folder_new.zip](https://github.com/Autodesk/maya-usd/files/11091947/Layout_Folder_new.zip)

**Comment by DGaggio** [comment]
Hi Natalia,

Thank you very much for your help, I've tested and it's working properly between my desktop and the laptop. I'll run a test tomorrow with my group mates as here it's a bit late.
I've tried to mess up a bit to understang better how it's working and try to link the layers inside folders to have everything more organize 
![folders](https://user-images.githubusercontent.com/115561317/228350999-2f73f930-980b-4507-9f8b-816786c7a4c7.png)
![code](https://user-images.githubusercontent.com/115561317/228351053-4b385578-6995-4b18-9bae-318d2da38c1d.png)

It's actually working as soon as I'm deleting the part of the code that link to a specific drive o folder before the scene.
![path](https://user-images.githubusercontent.com/115561317/228351132-07c8f5ae-119c-4a43-b636-cd42b035e23b.png)

The only problem is, if I'm not referencing everything in the Master layer, when I open Maya I can see the scene and the various usd load properly but I've a "Path not found" in the USD Layer Editor.
It looks weird to me so I don't understand if it's because I've to link everything only in the Master or it's a possible bug because the assets are loaded properly but the layer not.
![result](https://user-images.githubusercontent.com/115561317/228353771-8144c981-da9c-4c1e-954e-ae21aa9fb570.png)

I'm attaching the scene file if you want to investigate further.
If in a future review if it's possible to implement the possibility to load in relative path for everything (even the usd I'm referencing from the outliner) would be huge appreciate it, modify the code of every single usd layer for all the reference it's a bit painful and time consuming.

Thank you agein for the help.
Davide

[Scene.zip](https://github.com/Autodesk/maya-usd/files/11093684/Scene.zip)

**Comment by womanyoyoyo-adsk** [comment, marked as answer]
I fixed your paths in LayoutFolder.usd and AnimationFolder.usd. See attached. They needed to be relative paths (with periods at the front to indicate you are going back a directory). It's tricky doing this by hand - some knowledge about relative pathing is needed. We are actually currently working on being able to make everything relative. This post validates how much we do need this. Stay tuned!

Natalia 

[Scene_new.zip](https://github.com/Autodesk/maya-usd/files/11093992/Scene_new.zip)
